static/src/javascripts/lib/cookies.js

```diff
@@ -43,6 +48,10 @@ const addCookie = (
 ): void => {
     const expires = new Date();
 
+    if ([name, value].map(isValidCookieValue).some(_ => _ === false)) {
```

Comment:
if you did `if (isValidCookieValue(name) || isValidCookieValue(value))` you could save instantiating a new array and 2 loops though it. 

it might also make your intention clearer?

Reply:
or maybe allow `isValidCookieValue` to accept unlimited args? so `isValidCookieValue(name)` and `isValidCookieValue(name, value1, value2)` etc are valid? 

then it could return `false` early while looping through all args, `true` if nothing fails?

Reply:
gonna be honest, I'm personally a huge fan of `[].some` but i quite like the loop idea. Will give that a go!

Reply:
i guess `some` exits early too, but exiting straight after checking avoids having to check everything first, then analyzing the results as a whole

Reply:
yes, exactly!

Reply:
turns out `[].some` does short circuit (neat!) but also eslint really doesn't want me using `for` loops

Reply:
I would agree with @sndrs that this code is a bit dense: `[name, value].map(isValidCookieValue).some(_ => _ === false)`.

`if (isValidCookieValue(name) || isValidCookieValue(value))` is much easier to read.

I'm also wondering, we use the same logic to test if the name and value are valid. So maybe the `isValidCookieValue` function should be renamed to make explicit it is not about cookie values, but rather supported characters?